megatron/training.py

```diff
@@ -669,12 +723,27 @@ def training_log(loss_dict, total_loss_dict, learning_rate, iteration,
 
 def save_checkpoint_and_time(iteration, model, optimizer, opt_param_scheduler):
     timers = get_timers()
+    args = get_args()
     # Extra barrier is added to make sure
     # all ranks report the max time.
     timers('save-checkpoint', log_level=0).start(barrier=True)
-    save_checkpoint(iteration, model, optimizer, opt_param_scheduler)
+    if not args.use_tensorizer:
+        save_checkpoint(iteration, model, optimizer, opt_param_scheduler)
+    else:
+        save_checkpoint_tensorizer(iteration, model, optimizer, opt_param_scheduler)
     timers('save-checkpoint').stop(barrier=True)
-    timers.log(['save-checkpoint'])
+    if args.timing_file is not None:
```

Comment:
This looks like repeated copy-and-pasted code that can be converted into a function. The only diference is the `load_checkpoint` and `save_checkpoint` keys.